### PR TITLE
Update diagnostic.rs

### PR DESCRIPTION
**Pull request by Shreyan11**
Small change in if statement but should improve code readability

# Objective

- Describe the objective or issue this PR addresses.
- If you're fixing a specific issue, say "Fixes #X".

## Solution

- Describe the solution used to achieve the objective above.

---

## Changelog

> This section is optional. If this was a trivial fix, or has no externally-visible impact, you can delete this section.

- What changed as a result of this PR?
- If applicable, organize changes under "Added", "Changed", or "Fixed" sub-headings
- Stick to one or two sentences. If more detail is needed for a particular change, consider adding it to the "Solution" section
  - If you can't summarize the work, your change may be unreasonably large / unrelated. Consider splitting your PR to make it easier to review and merge!

## Migration Guide

> This section is optional. If there are no breaking changes, you can delete this section.

- If this PR is a breaking change (relative to the last release of Bevy), describe how a user might need to migrate their code to support these changes
- Simply adding new functionality is not a breaking change.
- Fixing behavior that was definitely a bug, rather than a questionable design choice is not a breaking change.

### CODE REVIEW

File: crates/bevy_diagnostic/src/diagnostic.rs [hunk 1]
```diff
@@ -45,9 +45,7 @@ impl Diagnostic {
     /// Add a new value as a [`DiagnosticMeasurement`]. Its timestamp will be [`Instant::now`].
     pub fn add_measurement(&mut self, value: f64) {
         let time = Instant::now();
-        if self.max_history_length > 1 {
-            if self.history.len() == self.max_history_length {
-                if let Some(removed_diagnostic) = self.history.pop_front() {
+        if self.max_history_length > 1 && self.history.len() == self.max_history_length && let Some(removed_diagnostic) = self.history.pop_front() {
```
Comment (djeedai) [inline crates/bevy_diagnostic/src/diagnostic.rs:48]: This code does not build. It also changes the logic by not performing the `self.sum -= ...` below under the same condition.

As a side note, I personally find it a lot _less_ readable as it crams 3 conditions on a single long line.

**Review by djeedai — CHANGES_REQUESTED** [review]

File: crates/bevy_diagnostic/src/diagnostic.rs (hunk 1 above)
Comment (tguichaoua) [inline crates/bevy_diagnostic/src/diagnostic.rs:48]: `if let chain` is an unstable feature : https://github.com/rust-lang/rust/issues/53667

You must use separate the `if let` from the other conditions.

Comment (Shreyan11) [reply crates/bevy_diagnostic/src/diagnostic.rs:48]: Very well

Comment (Shreyan11) [reply crates/bevy_diagnostic/src/diagnostic.rs:48]: Pardon me,my bad

Comment (Shreyan11) [reply crates/bevy_diagnostic/src/diagnostic.rs:48]: Pardon me,my bad

**Pull request closed without merging**